docs/designs/2967-package-image-vulnerability-scanning.md

```diff
@@ -0,0 +1,635 @@
+<!-- The sections in this template are highly-suggested, but optional. They exist to
+provide an idea of what an in-depth proposal **might** include. If a different
+format fits your proposal, feel free to use it, with the expectation that:
+
+1. You must link to the proposal issue (github) under the title
+1. If you omit sections that are relevant, you may be asked by maintainers to
+   add them in order to best review the proposal. -->
+
+# Image Vulnerability Scanning for Tanzu Packages
+
+* Proposal: [Issue #2967](https://github.com/vmware-tanzu/community-edition/issues/2967)
+* See Also:
+  * [k/k image scanning](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/vulnerability-mgmt/container-images.md)
+  * [k/k scanning job on testgrid](https://testgrid.k8s.io/sig-security-snyk-scan#ci-kubernetes-snyk-master)
+
+## Summary
+
+Create a periodically running prow job that:
+
+* Identifies all the images that are part of available tanzu packages
+* Runs a vulnerability scan on each image
+* Fails when an actionable vulnerability is found
+
+## Motivation
+
+Tanzu Community Edition is shipped prebaked with multiple packages that can be
+installed in a cluster. In line with having a secure supply chain, community
+edition maintainers lack visibility into container image vulnerabilities in
+these packages. Scanning them on a periodic basis will allow us to do just that.
+It will also help create a positive feedback loop where package owners will
+continually update their container images when new vulnerabilities are detected
+by us.
+
+### Goals
+
+* Identify a repeatable method to detect images that are part of default tanzu
+  packages
+* Run a vulnerability scan on these images, to detect fixable CVEs
+* Fail the prow job discretely based on pre-configured rules
+* Alert on failures to identified members of the community
+* Run the vulnerability scan as a periodically running job
+
```

Comment:
Are there non-goals of this proposal?

Reply:
Tried to use Future work as a replacement for non-goals. But perhaps vulnerability disclosure, embargo process falls in neither the goals nor future work. So I can add that as non-goals.